### PR TITLE
LimitHandlers Middleware

### PR DESCRIPTION
**Pull request by WyriHaximus**
LimitHandlers Middleware split off from #213
Assumes to be used with #215
Is blocked by #216 for the readme (will create merge conflicts otherwise)

Not sure about name, `LimitConcurrentHandlers` might be a better name, thoughts?

### CODE REVIEW

File: tests/Middleware/LimitHandlersTest.php [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        $this->assertTrue($calledC);
+    }
+
+    public function testStreamPauseAndResume()
+    {
+        $body = $this->getMockBuilder('React\Http\HttpBodyStream')->disableOriginalConstructor()->getMock();
+        $body->expects($this->once())->method('pause');
+        $body->expects($this->once())->method('resume');
+        $limitHandlers = new LimitHandlers(1);
+        $limitHandlers(new ServerRequest('GET', 'https://example.com/', [], $body), function () {});
```
Comment (jsor) [inline tests/Middleware/LimitHandlersTest.php]: `[]` -> `array()` for our old friend PHP 5.3

**Comment by jsor** [comment]
> Not sure about name, LimitConcurrentHandlers might be a better name, thoughts?

If find the `handlers` part a little bit unclear. Brain dump:

* LimitConcurrentRequests
* LimitConcurrentConnections
* LimitConcurrentClients

**Comment by WyriHaximus** [comment]
`Connections` isn't an option as the connection already has been made and is active, just pauzed when in the queue, same goes for clients tbh.

File: tests/Middleware/LimitHandlersTest.php (hunk 1 above)
Comment (WyriHaximus) [reply tests/Middleware/LimitHandlersTest.php]: Fixed -.-

**Comment by WyriHaximus** [comment]
@jsor what do you think about: `LimitConcurrentRequestHandlers`, a bit of a mouth full though :zipper_mouth_face:

**Comment by jsor** [comment]
@WyriHaximus That might work, yes. Maybe @clue has some more ideas.

**Comment by WyriHaximus** [comment]
@jsor currently stuck on the fact that the request stream pausing doesn't work as intended. When it's paused it will miss out on contents of the request body as shown by the functional test I've added. Hope @clue has some ideas for that more then the name at the moment 😛

**Comment by WyriHaximus** [comment]
Removed the mile stone because we hit a few roadblocks implementing this properly. Leaving this open for the time until we find a solution.

**Pull request closed without merging**